### PR TITLE
[torch.compile] support moe models

### PR DESCRIPTION
**Pull request by youkaichao**
moe models will read config file to determine the triton config to run.

reading files during forward is a disaster for `torch.compile` .

this pr wraps the config reading part inside a custom op, so that it can pass `torch.compile` (although `torch.compile` will not be able to optimize it).

### CODE REVIEW

File: tests/compile/test_basic_correctness.py [hunk 1]
```diff
@@ -13,11 +13,11 @@
 @pytest.mark.parametrize(
     "model, model_args, pp_size, tp_size, attn_backend, method, fullgraph",
     [
-        ("meta-llama/Llama-3.2-1B", [], 2, 2, "FLASH_ATTN", "generate", True),
+        ("meta-llama/Llama-3.2-1B", [], 2, 2, "FLASHINFER", "generate", True),
         ("nm-testing/Meta-Llama-3-8B-Instruct-W8A8-Dyn-Per-Token-2048-Samples",
          ["--quantization", "compressed-tensors"
           ], 1, 1, "FLASH_ATTN", "generate", True),
-        ("google/gemma-2-2b-it", [], 1, 2, "FLASHINFER", "generate", True),
+        ("ibm/PowerMoE-3b", [], 1, 2, "FLASH_ATTN", "generate", True),
```
Comment (youkaichao) [inline tests/compile/test_basic_correctness.py:20]: it does not work with `FLASHINFER` due to some head size config.

File: vllm/model_executor/layers/fused_moe/layer.py [hunk 2]
```diff
@@ -132,17 +138,18 @@ def forward_tpu(
...
-                         gating_output=router_logits,
-                         renormalize=renormalize)
+        return fused_moe_pallas(hidden_states=x,
+                                w1=layer.w13_weight,
+                                w2=layer.w2_weight,
+                                topk=top_k,
+                                gating_output=router_logits,
+                                renormalize=renormalize)
+
+    forward_native = forward_cuda
```
Comment (WoosukKwon) [inline vllm/model_executor/layers/fused_moe/layer.py:152]: What is this for?

Comment (youkaichao) [reply vllm/model_executor/layers/fused_moe/layer.py:152]: when we use inductor, we will use `forward_native` . this function needs to be implemented.

Comment (youkaichao) [reply vllm/model_executor/layers/fused_moe/layer.py:152]: you can also write a pytorch native implementation, but I doubt if inductor can optimize it. that's why I use `forward_cuda` as `forward_native`

Comment (youkaichao) [reply vllm/model_executor/layers/fused_moe/layer.py:152]: basically, `forward_native` is the function we compile when we use inductor.

Comment (WoosukKwon) [reply vllm/model_executor/layers/fused_moe/layer.py:152]: I see. I was just surprised because I didn't know that a class method can be defined this way

Comment (WoosukKwon) [reply vllm/model_executor/layers/fused_moe/layer.py:152]: So it's essentially the same as

```python
def forward_native(self, *args, **kwargs):
    return self.forward_cuda(*args, **kwargs)
```
?

Comment (youkaichao) [reply vllm/model_executor/layers/fused_moe/layer.py:152]: yes.

**Review by WoosukKwon — APPROVED** [review]
LGTM!

**Pull request merged**